stetho/src/main/java/com/facebook/stetho/inspector/database/SqliteDatabaseDriver.java

```diff
@@ -82,9 +80,20 @@ private void bootstrapNewPeer(JsonRpcPeer peer) {
       eventParams.database = databaseParams;
 
       peer.invokeMethod("Database.addDatabase", eventParams, null /* callback */);
```

Comment:
@jasta the `DatabaseDriver` is an abstract class now and all registration databases logic to DevTools is now part of this class and not in implementators. Which means that `peer.invokeMethod("Database.addDatabase", eventParams, null /* callback */);` is part of this class only and the subclasses don't know what is `JsonRpcPeer` is. Which is good I think. The method names was changes as you suggested and they are abstract. 

Regarding the `ExecuteResultHandler`, I'm not sure I understood you, but as it looks from code the generics can be removed at all in this interface. It could be strictly types like this: 
``` java
public interface ExecuteResultHandler {
  ExecuteSQLResponse handleRawQuery() throws SQLiteException;
  ExecuteSQLResponse handleSelect(Cursor result) throws SQLiteException;
  ExecuteSQLResponse handleInsert(long insertedId) throws SQLiteException;
  ExecuteSQLResponse handleUpdateDelete(int count) throws SQLiteException;
}
```
For this interface, I didn't make this change. Let me know if you want to remove the `<T>` at all. wdyt?